README.md

```diff
@@ -4,44 +4,62 @@
 
-## Define the Data Model
+### Implement the Data Model
 
-Create an `IOU` entity class with fields like `id`, `borrower`, `lender`, `amount`, and `date`
+Inside the `ious` package, create an `IOU` entity class with the following members:
+- `final UUID id` - unique identifier for the IOU
+- `String borrower` - name of borrower
+- `String lender` - name of lender
+- `BigDecimal amount` - amount covered by IOU
+- `Instant dateTime` - date and time of IOU being issued
+- `public IOU(String borrower, String lender, BigDecimal amount, Instant datetime)` - constructor that should generate a new UUID, use [the documentation][1] to learn how to generate a random value.
 
+Set the appropriate accessibility modifiers for all members and add appropriate property getters and setters.
+
+A representative HTTP response will look something like this:
 ```JSON
 {
-    "id": "",
+    "id": "5a6b7692-2322-482e-80cf-a59eedb9164f",
     "borrower": "Bob",
     "lender": "Alice",
     "amount": 1877.51,
     "date": "2023-04-23T18:25:43.511Z"
 }
 ```
 
-## Implement RESTful Endpoints 
+### Implement the Service
+
+A service interface, `IOUService` has been provided for you. Create a class that implements this interface using a `List` instance as the backing store and name it something sensible, e.g. `ListIOUService`, with the following members:
+
+- `final List<IOU> ious = new ArrayList<>();`
+
+Set the appropriate accessibility modifiers for all members and annotate the class as a [Service][2].
+
+### Implement the Controller
 
-Create RESTful endpoints for the following operations:
+Create an `IOUController` class with the following members:
 
-1. Retrieve a list of all IOUs (HTTP GET)
-1. Retrieve a specific IOU by ID (HTTP GET)
-1. Create a new IOU (HTTP POST)
-1. Update an existing IOU (HTTP PUT)
-1. Delete an IOU by ID (HTTP DELETE)
+- `IOUService iouService` - in instance of the service interface
+- `IOUController(IOUService iouService)` - constructor that accepts an instance of the service interface
 
-| Method | URL | Description | 
-|---|---|---|
-| `GET` | `/ious`| Get all IOUs |  
-| `GET` | `/ious/{id}`| Get an IOU by id |
-| `POST` | `/ious` | Add an IOU |  
-| `PUT` | `/ious/{id}` | Replace an IOU by Id |
-| `DELETE` | `/ious/{id}` | Delete an IOU by id |
+Add additional methods that defines endpoints for the following operations:
+
+| Method   | URL              | Description          |
+| -------- | ---------------- | -------------------- |
+| `GET`    | `/api/ious`      | Get all IOUs         |
+| `GET`    | `/api/ious/{id}` | Get an IOU by id     |
+| `POST`   | `/api/ious`      | Add an IOU           |
+| `PUT`    | `/api/ious/{id}` | Replace an IOU by Id |
+| `DELETE` | `/api/ious/{id}` | Delete an IOU by id  |
+
+Set the appropriate accessibility modifiers for all members and annotate the class as a [RestController][3].
 
 ## Learning Objectives:
```

Comment:
Should we add some details for how they can use Postman to interact with their solution once implemented? I imagine that some students trying to get clear on how APIs work might want to try it. 

Could be an example of a POST request to send a new IOU in the body and receive a response

Reply:
@sohbaker is this really the place for that though? As they'd already be introduced to Postman/REST clients via the course material?

Reply:
Perhaps an example of a response or a screenshot of Postman/REST clients? But if that's already in the slides, then that's good enough

Reply:
There aren't currently any examples of Postman in the slides, to my knowledge

Reply:
@Genyus Students are likely to come up against working with Postman to help verify their API (both for this exercise and later for their assessment projects). They'll need to use Postman to construct appropriate requests and see the returned responses.

As they'll want to refer to previous exercises to help them, it could be valuable to include some detail here. This way they can verify their implementation not only with tests but also by interacting with their REST service through Postman. An example of one request to get them started with that seems beneficial (whether a simple Postman screenshot or an example JSON body for a request).

Reply:
@sohbaker I'm still unconvinced that this is how and where Postman should be introduced. There are slides covering it for the [Node module](https://docs.google.com/presentation/d/1nGRWgHKA9qM6dd5FldcvbJhidbsiAXNwxYNByne7pa8/edit#slide=id.g102130ff34b_0_1) that could be easily adapted for the Java course and it feels more appropriate to present it properly in session as opposed to minimal coverage in a README.

Reply:
This is not the place to introduce a new concept or tooling. If Postman is the API testing tool of choice it will need to be introduced in the slides and reinforced here. 

[Slide 46](https://docs.google.com/presentation/d/1nGRWgHKA9qM6dd5FldcvbJhidbsiAXNwxYNByne7pa8/edit#slide=id.g102130ff34b_0_1) of the Node unit shows an example of introducing Postman, and [here's an example of it being reinforced in a README](https://github.com/cbfacademy/nodejs-starter-files/blob/bc35f7c497bbaedcf077a3305757c0966e1cc84b/node-server/README.md?plain=1#L74).

It sounds like changes need to be made to the slides, and a reminder to use the tool needs to be added to this file